x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -293,9 +295,15 @@ export function registerIndexRoutes({
     elasticsearchErrorHandler(log, async (context, request, response) => {
       const indexName = decodeURIComponent(request.params.indexName);
       const { client } = (await context.core).elasticsearch;
-      const pipelines = await getCustomPipelines(indexName, client);
+      const [defaultPipeline, customPipelines] = await Promise.all([
+        getPipeline(DEFAULT_PIPELINE_NAME, client),
```

Comment:
I went back and forth on if I should add a new endpoint for getting a pipeline or one specific to the default pipeline. But eventually landed on returning the default ingest pipeline from this endpoint we were already calling. This works for indices that do and do not already have custom pipelines.

We already have an endpoint for the default pipeline, but it's parsing it and returning a response specific to what the ingest pipeline card/modal uses via the `PipelinesLogic`. 

It _might_ be worthwhile to refactor the default pipeline endpoint to return the full `IngestPipeline` and then do the parsing in the `PipelinesLogic`. Then we would not need to return the default pipeline here. but instead combine the custom pipelines and default pipeline into a single list in the frontend logic.

Reply:
I like the refactor idea. Do you think we can handle this during the node architecture overhaul ?